### DISCUSSION TITLE
What's current limitation in implementing the spring constraint and other feature?

### DISCUSSION THREAD

**Discussion by ODtian** [godot-jolt/godot-jolt#447] [Questions] [answered]
Since godotphysics does not have spring constraint/joint yet, I was wondering if that was the point. Could we provide extra node or extend physics server api to gain the ability to access those powerful feature immediately?

**Comment by mihe** [comment, marked as answer]
There's work happening in #430 to add support for this to the existing nodes, which has been made possible only recently thanks to some very generous support from Jolt. I'm not sure exactly how much will be supported, but 6DOF springs are already working just fine in that pull request. I'm hoping to be done with that within the next 2-3 weeks.

Beyond that I have plans (see [the project board](https://github.com/orgs/godot-jolt/projects/1)) to add extra nodes that will better suit Jolt's constraints, which should hopefully be nicer to use than the default ones. This comes with some pretty major downsides right now though, which is that C# in Godot doesn't support custom nodes exposed by extensions, so I'm not entirely sure what to do there. It might be possible to create some custom bindings and expose that as a separate NuGet package.

**Reply by ODtian** [reply]
Really nice to hear that! I will try 6dof right away.

**Reply by mihe** [reply]
It only works in that pull request, so make sure you clone the [appropriate fork/branch](https://github.com/mihe/godot-jolt/tree/joint-params). I just merged some important bug fixes into it, so make sure you get those as well, if you happened to beat me to it.

**Reply by mihe** [reply]
6DOF springs are included in the newly released [0.4.0-rc1](https://github.com/godot-jolt/godot-jolt/releases/tag/v0.4.0-rc1).
